src/main/kotlin/net/casual/arcade/border/ArcadeBorder.kt

```diff
@@ -36,23 +43,37 @@ abstract class ArcadeBorder: WorldBorder() {
     }
 
     override fun getCenterX(): Double {
-        return this.centerX
+        return if (this.centerState != null) this.centerState.getCenter().x else 0.0
     }
 
     override fun getCenterZ(): Double {
-        return this.centerZ
+        return if (this.centerState != null) this.centerState.getCenter().y else 0.0
     }
 
+
     override fun setCenter(x: Double, z: Double) {
-        this.centerX = x
-        this.centerZ = z
+        //Somehow this is still null when this would need to be called to init
+        this.centerState = StillCenterBorderState(this.centerState.getCenter())
+
         this.state.onCenterChange()
+        this.centerState.onCenterChange()
```

Comment:
You don't need `onCenterChange` for the center state - after all it literally dictates when the center changes.